src/actions/setPost.tsx

```diff
@@ -0,0 +1,16 @@
+import { db } from '../firebase';
+
+interface FeedPostProps {
+  ownerUid: string | undefined | null;
+  displayName: string | undefined | null;
+  avatar?: string | undefined | null;
+  content: string;
+  timestamp: string;
+}
+
+export const setPost = async ({ ownerUid, content, avatar, timestamp, displayName }: FeedPostProps): Promise<void> => {
```

Comment:
I think you don't need to use async if you are not using await in your function.